src/utils.jl

```diff
@@ -56,13 +65,13 @@ always returning a random number generator.
 At least one of these arguments must be `nothing`.
 """
 function rng_from_rng_or_seed(rng::Union{Nothing, AbstractRNG}, seed::Union{Nothing, Integer})
-
     # TODO at some point we might emit a deprecation warning if a seed is specified
-
-    !(isnothing(seed) || isnothing(rng)) && throw(ArgumentError("Cannot specify both, seed and rng"))
```

Comment:
I think it is still better to not to allowing to set both, the `seed` and the `rng` argument - users might be confused if they set both but then the seed wont have any effect. Or do you have a specific reason why you would allow both?

Reply:
I wanted to be the least disruptive of the tests since there are a lot of line changes in this PR.
This is to retain compatibility with the tests setting an explicit seed (e.g. https://github.com/JuliaGraphs/Graphs.jl/blob/903f309039fa71ae2364aaec2158939f76cf5ccf/test/simplegraphs/generators/randgraphs.jl#L12).

IMO, I do think that giving an `rng` and a valid `seed` is unambiguous, meaning:
1. seed the given `rng` to `seed` (see https://github.com/JuliaGraphs/Graphs.jl/blob/903f309039fa71ae2364aaec2158939f76cf5ccf/src/utils.jl#L72);
2. return this `rng`.

Reply:
I still would not do it for the following reasons:
* We want to get rid of `seed` eventually, so that seeding the rng will fall to the callers side anyway.
* While `rng` gets mutated anyway, it is still not clear to the caller, that this what is happening.
* Not all subtypes of `rng` allow one to use `seed!`, one example is `RandomDevice`.

It would need much more code to get this running - in fact I implemented that in this PR based on this branch: `https://github.com/t-bltg/Graphs.jl/pull/1` -> for some reason that PR ended up in your fork and no in the main upstream repository though.

Reply:
Merged.